app/models/miq_queue.rb

```diff
@@ -237,7 +237,12 @@ def self.put_or_update(find_options)
     #   it's unordered nature.
     where_scope = if conds.key?(:args)
                     args = YAML.dump conds.delete(:args)
-                    MiqQueue.where(conds).where(['args = ?', args])
+                    if conds.key?(:miq_callback)
+                      miq_callback = YAML.dump conds.delete(:miq_callback)
+                      MiqQueue.where(conds).where(['args = ?', args]).where(['miq_callback = ?', miq_callback])
+                    else
+                      MiqQueue.where(conds).where(['args = ?', args])
```

Comment:
@kbrock The hash rocket probably doesn't work the same way because it breaks the tests.